presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoInstaller.java

```diff
@@ -85,21 +95,129 @@ private static void installUsingRpm(String pathToRpm, boolean checkDependencies)
+    private static void addTpchCatalog(Path directory)
             throws PrestoManagerException
     {
         try {
-            if (!isDirectory(CATALOG_DIR)) {
-                createDirectory(CATALOG_DIR);
-                write(CATALOG_DIR.resolve("tpch.properties"), "connector.name=tpch".getBytes());
+            if (!isDirectory(directory)) {
+                createDir(directory);
             }
+            write(directory.resolve("tpch.properties"), "connector.name=tpch".getBytes());
         }
         catch (IOException e) {
             throw new PrestoManagerException("Failed to add tpch catalog", e);
         }
     }
+
+    public static void installUsingTar(URL urlToFetchPackage)
+            throws PrestoManagerException
+    {
+        preInstallChecks();
+        File tarTempFile = fetchFileFromUrl(urlToFetchPackage);
+        try {
+            if (isDirectory(TAR_INSTALLATION_DIR)) {
+                deleteDirContents(TAR_INSTALLATION_DIR);
+            }
+            else {
+                createDir(TAR_INSTALLATION_DIR);
+            }
+            untarFile(tarTempFile, TAR_INSTALLATION_DIR);
+        }
+        finally {
+            if (!tarTempFile.delete()) {
+                LOGGER.warn("Failed to delete the temporary file: %s", tarTempFile.toString());
+            }
+        }
+        createDir(TAR_INSTALLATION_DIR.resolve("etc"));
+        addTpchCatalog(TAR_INSTALLATION_DIR.resolve("etc/catalog"));
+    }
+
+    private static void preInstallChecks()
+            throws PrestoManagerException
+    {
+        if (Float.valueOf("1.8").compareTo(Float.valueOf(System.getProperty("java.specification.version"))) < 0) {
+            throw new PrestoManagerException("Java version 8 or above is required");
+        }
+    }
+
+    public static void addConfigFiles()
+            throws PrestoManagerException
+    {
+        addConfigProperties();
+        addEnvSh();
+        addJvmConfig();
+        addNodeProperties();
+    }
+
+    private static void addConfigProperties()
+            throws PrestoManagerException
+    {
+        Properties properties = new Properties();
+        properties.setProperty("coordinator", "true");
+        properties.setProperty("node-scheduler.include-coordinator", "true");
+        properties.setProperty("http-server.http.port", "8080");
+        properties.setProperty("discovery-server.enabled", "true");
+        properties.setProperty("discovery.uri", "http://localhost:8080");
+        try (OutputStream output = newOutputStream(TAR_INSTALLATION_DIR.resolve("etc/config.properties"), CREATE_NEW)) {
+            properties.store(output, "single node install config");
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to add config.properties", e);
+        }
+    }
+
+    private static void addNodeProperties()
+            throws PrestoManagerException
+    {
+        // TODO: Add node.server-log-file & node.launcher-log-file
```

Comment:
All these add property methods could possibly be refactored as one method that takes a map with name -> value mapping